tests/integration/scripts/test_list.py

```diff
@@ -0,0 +1,340 @@
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Test the "cylc list" command."""
+
+import pytest
+
+from cylc.flow.exceptions import InputError
+from cylc.flow.option_parsers import Options
+from cylc.flow.scripts.list import (
+    get_option_parser,
+    _main,
+)
+
+
+ListOptions = Options(get_option_parser())
+
+
+@pytest.fixture(scope='module')
+async def cylc_list(mod_flow, mod_scheduler, mod_start):
+    id_ = mod_flow(
+        {
+            'scheduling': {
+                # NOTE: all "a*" tasks are in the graph, but not "b*" tasks
+                'initial cycle point': 1,
+                'cycling mode': 'integer',
+                'graph': {'P1': 'a12 & a111 & a112 & a121 & a2'},
+            },
+            'runtime': {
+                'A': {'meta': {'title': 'Title For A'}},
+                'A1': {'inherit': 'A'},
+                'A11': {'inherit': 'A1', 'meta': {'title': 'Title For A11'}},
+                'A12': {'inherit': 'A1'},
+                'a2': {'inherit': 'A'},
+                'a12': {'inherit': 'A1'},
+                'a111': {'inherit': 'A11'},
+                'a112': {'inherit': 'A11'},
+                'a121': {'inherit': 'A12'},
+                'B': {},
+                'b1': {'inherit': 'B'},
+            },
+        }
+    )
+    schd = mod_scheduler(id_)
+
+    async def _list(capsys, **kwargs):
+        nonlocal schd
+        capsys.readouterr()
+        await _main(ListOptions(**kwargs), schd.workflow)
+        out, err = capsys.readouterr()
+        return out.splitlines()
+
+    async with mod_start(schd):
+        yield _list
+
+
+@pytest.fixture
+def supports_utf8(monkeypatch):
+    monkeypatch.setenv('LANG', 'C.utf-8')
+
+
+@pytest.fixture
+def does_not_support_utf8(monkeypatch):
+    monkeypatch.setenv('LANG', 'C')
+
+
+async def test_plain(cylc_list, supports_utf8, capsys):
+    """Test the default output format."""
+    assert await cylc_list(capsys) == [
+        'a111',
+        'a112',
+        'a12',
+        'a121',
+        'a2',
+    ]
+
+    assert await cylc_list(capsys, all_tasks=True) == [
+        'a111',
+        'a112',
+        'a12',
+        'a121',
+        'a2',
+        'b1',  # <= in the runtime but not in the graph
+    ]
+
+    assert await cylc_list(capsys, all_namespaces=True) == [
+        'A',
+        'A1',
+        'A11',
+        'A12',
+        'B',
+        'a111',
+        'a112',
+        'a12',
+        'a121',
+        'a2',
+        'b1',
+        'root',
+    ]
+
+    with pytest.raises(InputError):
+        await cylc_list(capsys, all_tasks=True, all_namespaces=True)
+
+
+async def test_mro(cylc_list, supports_utf8, capsys):
+    """Test the --mro option."""
```

Comment:
I really don't like alphabetisms as options.

```suggestion
    """Test the --mro (method resolution order) option."""
```

Reply:
Point taken, but it's pretty niche and the long form is VERY long for an option name, so I went for explaining in the help text:
```
  -m, --mro             Print the linear "method resolution order" for each
                        namespace (the multiple-inheritance precedence order
                        as determined by the C3 linearization algorithm).
```
But feel free to put up a quick PR for `--method-resolution-order` if you like (not sure I'd bother, but no harm in having it).